crates/prelude/src/button.rs

```diff
@@ -66,13 +67,19 @@ impl<H: Handler> Listener<H> {
     /// ```ignore
     /// Listener::new(index, |state| debug!("Button has been {state:?}"))
     /// ```
-    pub fn new(button: usize, handler: H) -> Self {
+    #[track_caller]
+    pub fn new(button: usize, handler: H) -> Result<Self, Error> {
```

Comment:
There's a good design question here. Do we want to split error handling in the prelude? There are essentially 3 high-level categories:
- Never return error and always panic.
- Always return error and never panic.
- Return non-user error and panic on user error.

The first 2 have a signature that describe exactly what they do. Either the function returns a `Result` or not. The last one is more weird, because it returns a `Result` forcing the user to do something but still takes decision for them. So it's like a lose-lose situation, where the user loses the convenience of not dealing with errors but also loses the option to deal with its own errors. So I would suggest to keep prelude signatures to either of the first 2 alternatives (either always panic, or always return an error). It's not yet clear to me what's best between both and can be decided on a function basis. Ultimately I would expect return `Result<_, Error>` to be the best option and having the `main()` function return `Result<_, Error>` too, such that users can simply use `?` if they don't care about the error.

So in this particular case, I would suggest return a `Result` as you did (updating the documentation by adding `?`) and never panicking, letting the user handle all errors.

Reply:
I've made it such that we always return an error here as you suggested However, do you mind if I open another PR to have the `Result` be propagated up to `main`'s return value and leave the calls to `button::Listener::new` to unwrap for now? I think I'll have to touch many of the other examples that don't use `button::Listener::new`, as I'll be changing the function signature of main.

Reply:
> do you mind if I open another PR to have the `Result` be propagated up to `main`'s return value

Yes, that's actually what I had in mind.